packages/styled-components/src/models/StyledNativeComponent.js

```diff
@@ -173,42 +169,58 @@ export default (InlineStyle: Function) => {
      * forwardRef creates a new interim component, which we'll take advantage of
      * instead of extending ParentComponent to create _another_ interim class
      */
+    let WrappedStyledComponent;
 
-    // $FlowFixMe
-    WrappedStyledNativeComponent.attrs = finalAttrs;
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const forwardRef = (props, ref) => useStyledComponentImpl(WrappedStyledComponent, props, ref);
 
-    WrappedStyledNativeComponent.displayName = displayName;
+    forwardRef.displayName = displayName;
 
-    // $FlowFixMe
-    WrappedStyledNativeComponent.shouldForwardProp = shouldForwardProp;
+    // $FlowFixMe this is a forced cast to merge it StyledComponentWrapperProperties
+    WrappedStyledComponent = (React.forwardRef(forwardRef): StyledComponentWrapper<*, *>);
 
-    // $FlowFixMe
-    WrappedStyledNativeComponent.inlineStyle = new InlineStyle(
+    WrappedStyledComponent.attrs = finalAttrs;
+    WrappedStyledComponent.inlineStyle = new InlineStyle(
       // $FlowFixMe
       isTargetStyledComp ? target.inlineStyle.rules.concat(rules) : rules
     );
+    WrappedStyledComponent.displayName = displayName;
+    WrappedStyledComponent.shouldForwardProp = shouldForwardProp;
 
-    // $FlowFixMe
-    WrappedStyledNativeComponent.styledComponentId = 'StyledNativeComponent';
-    // $FlowFixMe
-    WrappedStyledNativeComponent.target = isTargetStyledComp
+    // this static is used to preserve the cascade of static classes for component selector
+    // purposes; this is especially important with usage of the css prop
+    WrappedStyledComponent.foldedComponentIds = isTargetStyledComp
+      ? // $FlowFixMe
+        Array.prototype.concat(target.foldedComponentIds, target.styledComponentId)
```

Comment:
I'm not sure why this is using `Array.prototype.concat` since `target.foldedComponentIds` should be known to be an array, so this can just become a `concat` method call.